AEPLifecycle/Tests/FunctionalTests/LifecycleFunctionalTests.swift

```diff
@@ -192,16 +193,20 @@ class LifecycleFunctionalTests: XCTestCase {
         mockRuntime.simulateComingEvents(startEvent1, pauseEvent, startEvent2)
 
         // verify
-        XCTAssertEqual(2, mockRuntime.dispatchedEvents.count)
+        XCTAssertEqual(5, mockRuntime.dispatchedEvents.count)
         XCTAssertEqual(2, mockRuntime.createdSharedStates.count)
-        XCTAssertEqual("1", (mockRuntime.dispatchedEvents[0].data?["lifecyclecontextdata"] as? [String: Any])?["launches"] as? String)
-        XCTAssertEqual("2", (mockRuntime.dispatchedEvents[1].data?["lifecyclecontextdata"] as? [String: Any])?["launches"] as? String)
-
-        XCTAssertEqual(1_595_909_459, mockRuntime.dispatchedEvents[1].data?["previoussessionstarttimestampmillis"] as? Double)
-        XCTAssertEqual(1_595_909_469, mockRuntime.dispatchedEvents[1].data?["previoussessionpausetimestampmillis"] as? Double)
-        XCTAssertEqual(86400.0 * 7.0, mockRuntime.dispatchedEvents[1].data?["maxsessionlength"] as? Double)
-        XCTAssertEqual(1_595_909_499, mockRuntime.dispatchedEvents[1].data?["starttimestampmillis"] as? Double)
-        XCTAssertEqual("start", mockRuntime.dispatchedEvents[1].data?["sessionevent"] as? String)
+        let dispatchedLifecycleStartEvent1 = mockRuntime.dispatchedEvents[1]
+        let dispatchedLifecycleStartEvent2 = mockRuntime.dispatchedEvents[4]
+
+
+        XCTAssertEqual("1", (dispatchedLifecycleStartEvent1.data?["lifecyclecontextdata"] as? [String: Any])?["launches"] as? String)
+        XCTAssertEqual("2", (dispatchedLifecycleStartEvent2.data?["lifecyclecontextdata"] as? [String: Any])?["launches"] as? String)
+
+        XCTAssertEqual(1_595_909_459, dispatchedLifecycleStartEvent2.data?["previoussessionstarttimestampmillis"] as? Double)
+        XCTAssertEqual(1_595_909_469, dispatchedLifecycleStartEvent2.data?["previoussessionpausetimestampmillis"] as? Double)
+        XCTAssertEqual(86400.0 * 7.0, dispatchedLifecycleStartEvent2.data?["maxsessionlength"] as? Double)
+        XCTAssertEqual(1_595_909_499, dispatchedLifecycleStartEvent2.data?["starttimestampmillis"] as? Double)
+        XCTAssertEqual("start", dispatchedLifecycleStartEvent2.data?["sessionevent"] as? String)
```

Comment:
We should either update this test to cover all events or split into two tests, one covering the regular events and the other covering the XDM events.